lib/common/output_xml.c

```diff
@@ -614,3 +614,15 @@ pcmk__output_xml_peek_parent(pcmk__output_t *out) {
     /* If queue is empty NULL will be returned */
     return g_queue_peek_tail(priv->parent_q);
 }
+
+void
+pcmk__output_set_legacy_xml(pcmk__output_t *out)
+{
+    private_data_t *priv = NULL;
+
+    CRM_ASSERT(out != NULL && out->priv != NULL);
+    CRM_CHECK(pcmk__str_eq(out->fmt_name, "xml", pcmk__str_none), return);
```

Comment:
If we make this a regular `if`, callers can call it unconditionally

Reply:
The "none" format has no private data, so the assertion for that would need to be moved past this check